Skip old PETSc (3.7.x or lower) in external petsc solver module

fdkong opened the pull request:
Closes #12733

<!--
INCLUDE THE FOLLOWING IN THE PR DESCRIPTION
- Explain relevant design information for your change.
- Follow the [Coding Standards](http://mooseframework.org/wiki/CodeStandards/).
- Submit or improve [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/).
- Reference a specific issue, place "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).
-->

fdkong committed:
Skip old PETSc (3.7.x or lower) in external petsc solver module

Closes #12733

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/278617/) on 82c14e7 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/12734/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/12734/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 03ee51349c9bc086e81b5e5f73b3325aa983b80a`

fdkong committed:
Combined-opt works with dynamic mode only

otherwise, we will have

Unable to locate library for "ExternalPetscSolverApp".
external_petsc_solver/test:external_petsc_problem.external_transient: We attempted to locate the library "libexternal_petsc_solver-opt.la" in the following paths:
external_petsc_solver/test:external_petsc_problem.external_transient:
external_petsc_solver/test:external_petsc_problem.external_transient:
external_petsc_solver/test:external_petsc_problem.external_transient: Make sure you have compiled the library and either set the "library_path" variable in your input file or exported "MOOSE_LIBRARY_PATH".
external_petsc_solver/test:external_petsc_problem.external_transient: Compiled in debug mode to see the list of libraries checked for dynamic loading methods.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/278705/) on 68c7260 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/12734/site)

This comment will be updated on new commits.
